New clang and added openjpeg2 in ci-common

aloysbaillet opened the pull request:
I've taken the helpful commit from @lgritz in #46 and packaged it up with a newer openjpeg2-devel package for OIIO.

Fixes #42 

I've already got clang built and released, I'm now rebuilding the OIIO and OCIO packages so everything should be available in the `aswftesting` images if anybody wants to try.

lgritz commented:
Can I ask for one more thing?
Also adding `giflib-devel` would really help me use the docker containers for OIIO.

aloysbaillet commented:
I added `giflib-devel` and also fixed a few packages that for some reason did not install properly in the first yum command... and I never noticed these issues!
All `aswftesting` images should be good to go, @lgritz would you be OK to try using them and tell me if they are working OK for OSL? I can then merge the PR to generate the `aswf` images.

lgritz commented:
I'll try it out today.

lgritz commented:
Yes, these seem fine, thanks!
